docs/rules/no-js-extension.md

```diff
@@ -36,6 +43,55 @@ define(function (require) {
     var foo = require('foo');
     /* ... */
 });
+
+require(['pluginname!foo'], function (foo) {
```

Comment:
Should this really be a warning though? It seems like the rule is overreaching here. Would it make more sense to have a separate rule for verifying that plugin-prefixed paths actually _have_ an extension?